### PR TITLE
Issue/13329 wizard cleanup

### PR DESCRIPTION
**Pull request by zwarm**
Parent #13329 

This PR cleans up the Backup Download wizard; focusing on removing unused code, handling onBackPressed situations and setting up intent extras for result. This is prep work for handling unhappy paths.

**Notes:**
- Styling has not been addressed
- The buttons are not hooked up, but you can tap on them and see a snackbar
- Testing for button clicks in progress/complete views will be added after the actions are hooked up.

**To Test**
CI Tests should pass

Do a quick run through the process and make sure the pages move through the process as expected
- Launch the app with BackupDownloadConfig flag on
- Navigate to ActivityLog
- Tap the More menu
- Tap Download Backup menu item
- Tap the Create downloadable file button
- The view changes to progress
- Note that the view changes to complete upon conclusion of process

PR submission checklist:

- [X] I have considered adding unit tests where possible.
- [X] I have considered adding accessibility improvements for my changes.
- [X] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/src/test/java/org/wordpress/android/ui/jetpack/backup/download/BackupDownloadViewModelTest.kt
Comment (ParaskP7) [inline WordPress/src/test/java/org/wordpress/android/ui/jetpack/backup/download/BackupDownloadViewModelTest.kt]: Minor (🔍): What do you think of the below alternative that utilised one `assertion` to achieve the same result?

```
assertThat(navigationTargets.last().wizardState)
                .isEqualTo(BackupDownloadState(rewindId = rewindId, downloadId = downloadId, published = published))
```

PS: Same applies to the below test. Although it already uses one `assertion` because it tried to verify only the `url` maybe it would be beneficial to keep them consistent. Wdyt?
PSS: Also, with this approach you are making sure that the `wizardState` is not being altered in any other way.

File: WordPress/src/test/java/org/wordpress/android/ui/jetpack/backup/download/BackupDownloadViewModelTest.kt
Comment (ParaskP7) [inline WordPress/src/test/java/org/wordpress/android/ui/jetpack/backup/download/BackupDownloadViewModelTest.kt]: 1) Question (❓): I am not sure if you want to utilise the `given, when, then` approach, but if you do want that, this test scenario could have been written as such: `given in detail step, when onBackPressed, then invokes wizard finished with cancel`

PS: The same apply to all tests. For example, the `then` wording is missing on all of them. Again, not sure if that is what you want. If not, please ignore this comment.

File: WordPress/src/test/java/org/wordpress/android/ui/jetpack/backup/download/BackupDownloadViewModelTest.kt
Comment (ParaskP7) [inline WordPress/src/test/java/org/wordpress/android/ui/jetpack/backup/download/BackupDownloadViewModelTest.kt]: Minor (🔍): Consider using the `BackupDownloadStep.DETAILS.id` instead of using zero directly. `0`, `1` and `2` can be considered magic numbers in this case.

PS: The same applies to the below two `Progress` and `Complete` related tests.

File: WordPress/src/test/java/org/wordpress/android/ui/jetpack/backup/download/BackupDownloadViewModelTest.kt
Comment (ParaskP7) [inline WordPress/src/test/java/org/wordpress/android/ui/jetpack/backup/download/BackupDownloadViewModelTest.kt:170]: Minor (🔍): Consider adding an empty line here to split the `Act` from the `Assert`.

File: WordPress/src/test/java/org/wordpress/android/ui/jetpack/backup/download/BackupDownloadViewModelTest.kt
Comment (ParaskP7) [inline WordPress/src/test/java/org/wordpress/android/ui/jetpack/backup/download/BackupDownloadViewModelTest.kt]: Minor (🔍): Consider splitting this to 3 different tests.

File: WordPress/src/test/java/org/wordpress/android/ui/jetpack/backup/download/BackupDownloadViewModelTest.kt
Comment (ParaskP7) [inline WordPress/src/test/java/org/wordpress/android/ui/jetpack/backup/download/BackupDownloadViewModelTest.kt]: Minor (🔍): Consider adding empty lines to split the `Arrange`, `Act` and `Assert`.

File: WordPress/src/test/java/org/wordpress/android/ui/jetpack/backup/download/BackupDownloadViewModelTest.kt
Comment (ParaskP7) [inline WordPress/src/test/java/org/wordpress/android/ui/jetpack/backup/download/BackupDownloadViewModelTest.kt:217]: Minor (🔍): Consider removing the explicit `savedInstanceState` named argument. Within this whole test suite, I am seeing both, this to be and not being used, most of the tests are not using it. This way all test will end up being more consistent. Also, due to the fact that this function accepts only one parameter it makes it less necessary.

**Review by ParaskP7 — APPROVED** [review]
👋 @zwarm !

I have tested the solution and it works as expected and described.

I am approving this PR since all my comments are test related. I will not merge this PR to give you some space to address any of the comments you feel addressing. Feel free to merge the PR afterwards or re-requesting my review to have another look.

File: WordPress/src/test/java/org/wordpress/android/ui/jetpack/backup/download/BackupDownloadViewModelTest.kt
Comment (zwarm) [reply WordPress/src/test/java/org/wordpress/android/ui/jetpack/backup/download/BackupDownloadViewModelTest.kt]: Sounds good, thanks for the suggestion.

File: WordPress/src/test/java/org/wordpress/android/ui/jetpack/backup/download/BackupDownloadViewModelTest.kt
Comment (zwarm) [reply WordPress/src/test/java/org/wordpress/android/ui/jetpack/backup/download/BackupDownloadViewModelTest.kt]: I split these out based on your earlier suggestion. :)

**Comment by zwarm** [comment]
@ParaskP7 - I incorporated your suggestions, want to take a second look and approve if all is ok? Thanks. :)

**Comment by ParaskP7** [comment]
> @ParaskP7 - I incorporated your suggestions, want to take a second look and approve if all is ok? Thanks. :)

Awesome @zwarm ! 🌟 I check and resolved of all them, thanks for incorporating these suggestions! ❤️ I am now going to merge this PR.

PS: Just a quick clarification on the `given` part of a test sentence. The `given` is optional and only required when there need to be an explicit statement to showcase the requirement or when there are multiple test that need to be differentiated by it. Only the `when` and `then` are always mandatory. For example:
- `given view model` is not needed, however
- `given in progress step` is indeed totally valid.

**Pull request merged**
